CHANGELOG.md

```diff
@@ -1,49 +1,229 @@
+  - Use `rustc-hash` to speed up resolver
+    [#14663](https://github.com/rust-lang/cargo/pull/14663)
+  - Share conflict cache between activation retries
+    [#14692](https://github.com/rust-lang/cargo/pull/14692)
+  - Avoid cloning when iterating using `RcVecIter`
+    [#14690](https://github.com/rust-lang/cargo/pull/14690)
+- Updated to `pulldown-cmark` 0.12.0 
+  [#14668](https://github.com/rust-lang/cargo/pull/14668)
+
 ## Cargo 1.83 (2024-11-28)
-[8f40fc59...HEAD](https://github.com/rust-lang/cargo/compare/8f40fc59...HEAD)
+[8f40fc59...rust-1.83.0](https://github.com/rust-lang/cargo/compare/8f40fc59...rust-1.83.0)
 
 ### Added
 
+- `--timings` HTML output can now auto-switch between light and dark color
+  schemes based on browser preference.
+  [#14588](https://github.com/rust-lang/cargo/pull/14588)
+- Introduced a new `CARGO_MANIFEST_PATH` environment variable,
+  similar to `CARGO_MANIFEST_DIR` but pointing directly to the manifest file.
+  [#14404](https://github.com/rust-lang/cargo/pull/14404)
+- manifest: Added `package.autolib`, allowing `[lib]` auto-discovery to be disabled.
+  [#14591](https://github.com/rust-lang/cargo/pull/14591)
+
 ### Changed
 
-- Enhanced the missing feature error message.
+- ❗️ Lockfile format v4 is now the default for creating/updating a lockfile.
+  Rust toolchains 1.78+ support lockfile v4.
+  For compatibility with earlier MSRV,
+  consdier setting the `package.rust-version` to 1.82 or earlier.
+  [#14595](https://github.com/rust-lang/cargo/pull/14595)
+- cargo-package: Now fails fast if the package version is already published.
+  [#14448](https://github.com/rust-lang/cargo/pull/14448)
+- Improved error messages for missing features.
   [#14436](https://github.com/rust-lang/cargo/pull/14436)
-- Enhanced the dependency update status message, including
-  - Reports incompatible packages with precise Rust version
-    [#14457](https://github.com/rust-lang/cargo/pull/14457)
-  - Reports incompatible-with-rustc when MSRV-resolver is disabled
-    [#14459](https://github.com/rust-lang/cargo/pull/14459)
-  - Reports compatible, incompatible, and direct-dep updates with different
-    colors and messages.
-    [#14461](https://github.com/rust-lang/cargo/pull/14461)
-  - The `Locking` status message stops showing workspace members.
-    [#14445](https://github.com/rust-lang/cargo/pull/14445)
 - Log details of `rustc` invocation failure if no errors are seen
   [#14453](https://github.com/rust-lang/cargo/pull/14453)
-- cargo-package: Don't automatically include the current crate when packaging
-  [#14488](https://github.com/rust-lang/cargo/pull/14488)
+- Uplifted `windows-gnullvm` import libraries, aligning them with `windows-gnu`.
+  [#14451](https://github.com/rust-lang/cargo/pull/14451)
+- Suggest `cargo info` command in the `cargo search` result
+  [#14537](https://github.com/rust-lang/cargo/pull/14537)
+- Improved error reporting when a feature is not found in `activated_features`.
```

Comment:
Using a variable name in the public facing changes seems odd

Reply:
While this panic message is public facing, this seems more like an internal/nightly stuff. AFAIK only things like `-Zbindeps` hit this and the message still exposes internal debug representation.

Moved to under internals.